great_expectations/data_context/store/_store_backend.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import uuid
+from abc import ABCMeta, abstractmethod
+from typing import Any, Optional
+
+import pyparsing as pp
+
+from great_expectations.exceptions import InvalidKeyError, StoreBackendError, StoreError
+
+logger = logging.getLogger(__name__)
+
+
+class StoreBackend(metaclass=ABCMeta):
+    """A store backend acts as a key-value store that can accept tuples as keys, to abstract away
+    reading and writing to a persistence layer.
+
+    In general a StoreBackend implementation must provide implementations of:
+      - _get
+      - _set
+      - list_keys
+      - _has_key
+    """
+
+    IGNORED_FILES = [".ipynb_checkpoints"]
+    STORE_BACKEND_ID_KEY = (".ge_store_backend_id",)
+    STORE_BACKEND_ID_PREFIX = "store_backend_id = "
+    STORE_BACKEND_INVALID_CONFIGURATION_ID = "00000000-0000-0000-0000-00000000e003"
+
+    def __init__(
+        self,
+        fixed_length_key=False,
+        suppress_store_backend_id=False,
+        manually_initialize_store_backend_id: str = "",
+        store_name="no_store_name",
+    ) -> None:
+        """
+        Initialize a StoreBackend
+        Args:
+            fixed_length_key:
+            suppress_store_backend_id: skip construction of a StoreBackend.store_backend_id
+            manually_initialize_store_backend_id: UUID as a string to use if the store_backend_id is not already set
```

Comment:
This defaults to "" which isn't a UUID. This might be better as an `Optional[uuid.UUID]` if being a UUID is important or you could do validation here if you need to keep this as a `str` argument.